cryptoservice/crypto_service.go

```diff
@@ -110,28 +100,23 @@ func (ccs *CryptoService) Sign(keyIDs []string, payload []byte) ([]data.Signatur
 			}
 		}
 
-		algorithm := privKey.Algorithm()
 		var sigAlgorithm data.SigAlgorithm
-		var sig []byte
 
-		switch algorithm {
-		case data.RSAKey:
-			sig, err = rsaSign(privKey, payload)
+		switch privKey.(type) {
+		case *data.RSAPrivateKey, *data.RSAx509PrivateKey:
```

Comment:
There should be no such thing as an `RSAx509PrivateKey`, only `RSAx509PublicKey`, correct?